library/std/src/fs.rs

```diff
@@ -1393,6 +1417,27 @@ impl Permissions {
     /// This operation does **not** modify the filesystem. To modify the
     /// filesystem use the [`set_permissions`] function.
     ///
+    /// Warning: `set_readonly(false)` makes the file *world-writable* on Unix.
+    /// You can use the [`PermissionsExt`] trait on Unix to avoid this issue.
+    ///
+    /// ## Windows
+    ///
+    /// On Windows this sets or clears [`FILE_ATTRIBUTE_READONLY`](https://docs.microsoft.com/en-us/windows/win32/fileio/file-attribute-constants).
+    /// If `FILE_ATTRIBUTE_READONLY` is set then writes to the file will fail
+    /// but the user may still have permission to change this flag.
+    /// The behaviour of this attribute for directories depends on the Windows
+    /// version.
```

Comment:
Perhaps it would be good to be a little less vague here (and above). The only thing setting the readonly attribute on a directory does is prevent deleting an empty directory until the flag is unset (also note that it doesn't prevent changing the directory contents). This hasn't changed exactly but `DeleteFile` will now automatically unset the flag itself before deleting the directory (assuming it has permission to write attributes).

Reply:
Ah to be honest I was mostly vague because I could not figure out exactly what it did or how it had changed. Microsoft's documentation is very unclear and the experiments I did on Windows didn't really match up with how it said it was supposed to behave (I can't remember the details exactly). Anyway I'll go with your description. :-)

Reply:
Do you have a reference for that btw? The only documentation I can find says it is ignored (which I guess is indistinguishable really).

Reply:
Unfortunately my only reference is "experience" which isn't a great answer. Searching the web the closest I can currently find is [a comment on SO](https://stackoverflow.com/questions/51767056/create-directory-without-read-only-attribute-on-windows-in-java#comment90711635_51767056). While I wouldn't normally consider that authoritative, the author is a prominent contributor to the Python standard library so does know their stuff.

Though that comment was made before the changes that made it do nothing.